M/mlpack/build_tarballs.jl

```diff
@@ -6,22 +6,21 @@ using BinaryBuilder
 
 # Set sources and other environment variables.
 name = "mlpack"
-version = v"3.4.3"
-source_version = v"3.4.2"
+version = v"4.0.0"
+source_version = v"4.0.0"
```

Comment:
I'd do
```suggestion
source_version = v"4.0.0"
version = source_version
```
so that in the future we can update a single line, if we don't need to diverge from the upstream version number.